### PR TITLE
Add `DeleteTagCommand`

### PR DESCRIPTION
**Pull request by malwaregarry**
#37 
Added the command to delete tags in Rapportbook.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteTagCommand.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public static final String COMMAND_SPECIFIER = "delete";
+    public static final String COMMAND_WORD = COMMAND_GROUP + " " + COMMAND_SPECIFIER;
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes tags with the names given. "
+            + "Parameters: "
+            + "[TAG1] [TAG2] ... \n"
+            + "Example: " + COMMAND_WORD + " tag1 tag2 tag3";
+
+    public static final String MESSAGE_SUCCESS = "Tags deleted: %1$s";
+    public static final String MESSAGE_DUPLICATE_TAGS = "Tag(s) %1$s does not exist in Rapportbook!";
```
Comment (kelvinou01) [inline src/main/java/seedu/address/logic/commands/DeleteTagCommand.java:26]: I think `MESSAGE_TAGS_DONT_EXIST` would be a more suitable name

File: src/main/java/seedu/address/logic/commands/CreateTagCommand.java [hunk 2]
```diff
@@ -21,10 +21,10 @@ public class CreateTagCommand extends TagCommandGroup {
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Creates tags with the names given. "
             + "Parameters: "
             + "[TAG1] [TAG2] ... \n"
-            + "Example: " + COMMAND_WORD + " tag1 tag2 tag3";
+            + "Example: " + COMMAND_WORD + COMMAND_SPECIFIER + " tag1 tag2 tag3";
```
Comment (kelvinou01) [inline src/main/java/seedu/address/logic/commands/CreateTagCommand.java:24]: Good catch!

File: src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java [hunk 3]
```diff
@@ -40,12 +44,28 @@ public TagCommandGroup parse(String args) throws ParseException {
         switch (commandSpecifier) {
         case CreateTagCommand.COMMAND_SPECIFIER:
             String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
-            return new CreateTagCommandParser().parse(String.join(" ", argsToPass));
```
Comment (kelvinou01) [inline src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java:43]: I think we should keep the `CreateTagCommandParser` and `DeleteTagCommandParser` here, to maintain a single layer of abstraction within this `switch` block?

**Review by kelvinou01 — CHANGES_REQUESTED** [review]
Great job finishing this feature in such a small timespan!

File: src/main/java/seedu/address/logic/commands/CreateTagCommand.java [hunk 4]
```diff
@@ -21,10 +21,10 @@ public class CreateTagCommand extends TagCommandGroup {
...
 
     public static final String MESSAGE_SUCCESS = "New tags created: %1$s";
-    public static final String MESSAGE_DUPLICATE_TAGS = "Tag(s) %1$s already exist in the address book";
+    public static final String MESSAGE_DUPLICATE_TAGS = "Tag(s) %1$s already exist in Rapporrbook!";
```
[the remark this answers is not in the document]
Comment (malwaregarry) [reply src/main/java/seedu/address/logic/commands/CreateTagCommand.java:27]: Yup, my bad

File: src/main/java/seedu/address/logic/commands/DeleteTagCommand.java (hunk 1 above)
Comment (malwaregarry) [reply src/main/java/seedu/address/logic/commands/DeleteTagCommand.java:26]: you're right. i forgot to change it after copy-pasting

File: src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java (hunk 3 above)
Comment (malwaregarry) [reply src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java:43]: I change it out as I didn't want to duplicate code. I didn't think of SLAP when I changed it. Perhaps I could abstract the creation into another function?
```java
switch (commandSpecifier) {
        case CreateTagCommand.COMMAND_SPECIFIER:
            return parseCreateTagCommand(argArray);
        case DeleteTagCommand.COMMAND_SPECIFIER:
            return parseDeleteTagCommand(argArray);
        case RemoveTagCommand.COMMAND_SPECIFIER:
            return parseRemoveTagCommand(argArray);
        default:
            return new TagCommandParser().parse(trimmedArgs);
        }
    }

private CreateTagCommand parseCreateTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new CreateTagCommand(parseArgs(argsToPass));
}

private DeleteTagCommand parseDeleteTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new DeleteTagCommand(parseArgs(argsToPass));
}

private RemoveTagCommand parseRemoveTagCommand(String[] argArray) throws ParseException {
    String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
    return new RemoveTagCommandParser().parse(String.join(" ", argsToPass));
}
```

Comment (Tex-Tang) [reply src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java:43]: Perhaps we can do
```
String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
switch (commandSpecifier) {
case CreateTagCommand.COMMAND_SPECIFIER:
    return new CreateTagCommand(parseArgs(argsToPass));
case DeleteTagCommand.COMMAND_SPECIFIER:
    return new DeleteTagCommand(parseArgs(argsToPass));
case RemoveTagCommand.COMMAND_SPECIFIER:
    return new RemoveTagCommandParser().parse(String.join(" ", argsToPass));
default:
    return new TagCommandParser().parse(trimmedArgs);
}
```

Comment (malwaregarry) [reply src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java:43]: "we"??

> 



> ```java
> String[] argsToPass = Arrays.copyOfRange(argArray, 1, argArray.length);
> ```
I thought of this as well. Wouldn't there be an error if the array is only size 1?

Comment (Tex-Tang) [reply src/main/java/seedu/address/logic/parser/TagCommandGroupParser.java:43]: As below,
image

**Comment by Polygonalr** [comment]
Aside from the changes others has requested, LGTM!

**Review by weiquany — COMMENTED** [review]
LGTM

**Pull request merged**